### PR TITLE
:zero: call drain on empty queues

### PR DESCRIPTION
**Pull request by markysoft**
Drain was not being called if an empty list was provided to a queue.
This left the ETL in an inconsistent state where cleanup had not occurred.

### CODE REVIEW

File: app/lib/etl.js [hunk 1]
```diff
@@ -88,7 +92,8 @@ function start() {
   });
 }
 
-function safeStart() {
+function safeStart(callback) {
+  completeCallback = callback;
```
Comment (markysoft) [inline app/lib/etl.js]: I needed to add a callback for testing purposes, I couldn't find an elegant way to signal the end of the ETL.

File: app/lib/config.js [hunk 2]
```diff
@@ -1,4 +1,7 @@
 const config = {
+  app: {
```
Comment (st3v3nhunt) [inline app/lib/config.js]: Given there is only a single instance of the app's name being used I don't see the rationale for adding it to the config.

File: test/integration/etl.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  it('should complete ETL if one page of results', (done) => {
+    stubOnePageOfResults();
+    etl.start(done);
+  });
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+});
+
```
Comment (st3v3nhunt) [inline test/integration/etl.js]: Remove this empty line please.

File: test/integration/etl.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+const etl = require('../../app/lib/etl');
+const service = require('../../app/lib/syndicationService');
+
+function readFile(path) {
+  return fs.readFileSync(path, 'utf8');
+}
+
+const uri = `/all.xml?apikey=${process.env.SYNDICATION_API_KEY}&page=1`;
+
+function stubNoResults() {
```
Comment (st3v3nhunt) [inline test/integration/etl.js]: Can this function be combined with `stubOnePageOfResults` if `filePath` is added as a parameter?

File: app/lib/etl.js (hunk 1 above)
Comment (st3v3nhunt) [reply app/lib/etl.js]: I wonder if the tests could await the result instead? http://mochajs.org/#using-async--await

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
+0.1.3 / 2017-10-24
```
Comment (st3v3nhunt) [inline CHANGELOG.md]: This will need updating if it doesn't get released tomorrow.

File: app/lib/queues/populateIdListQueue.js [hunk 6]
```diff
@@ -60,12 +60,15 @@ function addPageToQueue(q, pageNo) {
 }
 
 function start(totalPages, workers, drain) {
-  const q = async.queue(processQueueItem, workers);
+  if (totalPages > 0) {
```
Comment (st3v3nhunt) [inline app/lib/queues/populateIdListQueue.js]: I know there are no unit tests for this module atm but would it be possible to add some?

Comment (markysoft) [reply app/lib/queues/populateIdListQueue.js]: Have added some tests to improve coverage

**Pull request merged**